libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -580,3 +580,23 @@ AP_GPS::send_mavlink_gps2_rtk(mavlink_channel_t chan)
         drivers[1]->send_mavlink_gps_rtk(chan);
     }
 }
+
+// is this GPS instance configured
+bool AP_GPS::is_configured(uint8_t instance) const
+{
+    if (drivers[instance] != NULL && state[instance].status != NO_GPS) {
```

Comment:
Iff the GPS_TYPE is set appropriately then sure. But you don't check the type, you check the drivers[] which will have a null in it for a GPS_TYPE none, and then you would return false. (A non existent GPS should be considered configured). Or am I stuffing it all up and should go make a cup of coffee before looking at code this morning?